core/tests/puppeteer-acceptance-tests/user-utilities/blog-post-admin-utils.js

```diff
@@ -212,6 +238,28 @@ module.exports = class e2eBlogPostAdmin extends baseUser {
     }, blogPostTitle);
   }
 
+  /**
+   * This function checks that the user is unable to publish a blog post.
+   */
+  async expectUserUnableToPublishBlogPost() {
+    const toastMessageBox = await this.page.$(
+      'div.e2e-test-toast-warning-message');
+    const toastMessageWarning = await this.page.evaluate(
+      el => el.textContent, toastMessageBox);
+    const isPublishButtonDisabled = await this.page.$eval(
+      publishBlogPostButton, (button) => {
+        return button.disabled;
+      });
+    if (!isPublishButtonDisabled) {
+      throw new Error('User is able to publish duplicate blog post');
+    }
+    if (toastMessageWarning === duplicateBlogPostWarningMessage) {
```

Comment:
This isn't the way to do it. Why not just include the contents of toastMessageWarning in your showMessage() call?

Other than that, the previous approach of passing the expected warning message in and checking it against toastMessageWarning is still correct, and allows a stricter test.

Reply:
@seanlip Please take a look to the changes now!